### ISSUE TITLE
Error 400: invalid_request when authenticating with Google

### ISSUE THREAD

**Issue by WKHarmon** [iMicknl/ha-nest-protect#105]
### The problem

I installed the ha-nest-protect integration through HACS and restarted. I'm trying to add the integration now and when I click on the "authorize your account" link I get this error:

<img width="458" alt="Screen Shot 2022-10-07 at 12 20 03 PM" src="https://user-images.githubusercontent.com/245099/194638463-1f51bfda-dbca-407c-b678-93d786a191fc.png">

Clicking into the error details shows the following:

Error 400: invalid_request
The out-of-band (OOB) flow has been blocked in order to keep users secure. Follow the Out-of-Band (OOB) flow Migration Guide linked in the developer docs below to migrate your app to an alternative method.
Request details: access_type=offline response_type=code redirect_uri=urn:ietf:wg:oauth:2.0:oob client_id=<redacted> scope=openid https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/nest-account

I've tried it on two separate browsers to the same result.

Edited to add: I found this page: https://developers.google.com/identity/protocols/oauth2/resources/oob-migration

I was successfully able to get a token by appending a `ack_oob_shutdown=2022-10-03` parameter to the query string on the auth url.

### What version of this integration (ha-nest-protect) has the issue?

0.3.8

### What version of Home Assistant Core has the issue?

2022.10.0

### Device / Model

Nest Protect (Battery)

**Comment by WKHarmon** [comment]
I found this page: https://developers.google.com/identity/protocols/oauth2/resources/oob-migration

I was successfully able to get a token by appending a `ack_oob_shutdown=2022-10-03` parameter to the query string on the auth url.

**Comment by MarElemans** [comment]
Thanks! Worked for me

**Comment by kcline** [comment]
You're solution worked for me too, just commenting to say thanks and add that you need to append &ack_oob_shutdown=2022-10-03 to the url provided in the setup steps (right click the link from HA and copy url, append the command to the link, then paste it into a browser).

I'm definitely not a programmer, more of a tinker'er, so this took me a bit longer to remember to add the & to properly append it.  Mostly commenting in case others in my situation (and knowledge level) come across this issue before it's fixed.

**Comment by iMicknl** [comment]
I will have to fix this in the component indeed; however it worries me that this will eventually be deprecated.

**Comment by pacifer93** [comment]
I just tried it myself and got the same error but my solution was to press use another account then just use the same google account and it worked. (it gives out a warning about soon it will be deprecated) but yeah quick fix..

**Comment by semple1990** [comment]
Just highlighting kcline's comment about adding the & as i missed that briefly and wondered why it wasn't working.

So the authorization url should end in `...apps.googleusercontent.com`

You'll want to add `&ack_oob_shutdown=2022-10-03` to the end to make:

`...apps.googleusercontent.com&ack_oob_shutdown=2022-10-03`

**Comment by FlyersPh9** [comment]
Appending `&ack_oob_shutdown=2022-10-03` to the end of the URL copied from in HA wasn't working for me.  I instead opened the link incognito and was able to get my access token that way.

**Comment by jpearce73** [comment]
> Just highlighting kcline's comment about adding the & as i missed that briefly and wondered why it wasn't working.
> 
> So the authorization url should end in `...apps.googleusercontent.com`
> 
> You'll want to add `&ack_oob_shutdown=2022-10-03` to the end to make:
> 
> `...apps.googleusercontent.com&ack_oob_shutdown=2022-10-03`

I am getting something slightly different out of the UI link. I already have an extra parameter. My URL looks like this
https://.............apps.googleusercontent.com&flowName=GeneralOAuthFlow
Perhaps because I have a few Google accounts? 
The reply gives me this page
![image](https://user-images.githubusercontent.com/19938737/196051822-cbab88f8-5dee-48b8-946b-cd507451676c.png)

Continuing with my chosen ID fails with the message shown at the top of this thread,  whether I append '&ack_oob_shutdown=2022-10-03' or not

**Comment by redford-jones** [comment]
The workaround above appears to have stopped working, at least for me. I consistently get a 400 despite adding the query string shown above.

**Comment by tre4b** [comment]
I too have this issue, exactly the same as the OP.

One thing I wondered is whether this is down to the fact that the google account I am using is not my default account.  That one is a GSuite account and Nest does not work with those so I had to create an individual account.   I am logged into that account when I attempt to link the integration but get the message above...

**Comment by hallidaydaniel** [comment]
> > Just highlighting kcline's comment about adding the & as i missed that briefly and wondered why it wasn't working.
> > So the authorization url should end in `...apps.googleusercontent.com`
> > You'll want to add `&ack_oob_shutdown=2022-10-03` to the end to make:
> > `...apps.googleusercontent.com&ack_oob_shutdown=2022-10-03`
> 
> I am getting something slightly different out of the UI link. I already have an extra parameter. My URL looks like this https://.............apps.googleusercontent.com&flowName=GeneralOAuthFlow Perhaps because I have a few Google accounts? The reply gives me this page ![image](https://user-images.githubusercontent.com/19938737/196051822-cbab88f8-5dee-48b8-946b-cd507451676c.png)
> 
> Continuing with my chosen ID fails with the message shown at the top of this thread, whether I append '&ack_oob_shutdown=2022-10-03' or not

yes, I get the same behaviours as this now sadly - not managed to find work around

**Comment by idroyHA** [comment]
Also the same issue. With &ack_oob_shutdown=2022-10-03 still a 400 error.

**Comment by benceh90** [comment]
The same problem here. OOB is outdated :(

**Comment by Tatey53** [comment]
Yes, I can confirm that I get the same 400 error, and adding &ack_oob_shutdown=2022-10-03 gets me past it, but after selecting my account if goes back to the same 400 error
![error](https://user-images.githubusercontent.com/113981314/197030141-a96c97b7-cf95-4d18-931b-5846e590987b.png)

**Comment by pmkruyen** [comment]
After I encountered this problem and solved it(?), I read this thread. 

Original approach
- made new google account
- Set up nest account in app (ios)
- Ran login.js in Homebridge
- —> error

Solution:
- Logged in on Nest portal, had to give Nest access to my account
- Started a ne sesion of homebridge in incognito browser
- Ran login.js
- —> Solves

(Don’t know if the plugin works now, because I haven’t adopted any Nest devices yet)


Another thing is that I got a warning that this access method will be depreciated soon.

**Comment by simtreti** [comment]
@iMicknl Will you update the repo to correct that issue ?

**Comment by iMicknl** [comment]
@simtreti eventually, yes. However my time currently is very scarce, so I am open to PR's.

I am not a fan of the cookie method, I have to say, since this is not an easy method for many users. I will need to say if there can be a workaround for this issue, but that will take time.

**Comment by simtreti** [comment]
@iMicknl Take your time it was just to know if I need to check elsewhere to integrate my new nest protect into HA.

**Comment by m3tc0n** [comment]
I have the same problem. Hopefully this will be resolved soon.

**Comment by sadicarnot** [comment]
Before I got into Home Assistant, I had gotten the Starling Home Hub and I was able to get the Protects into Home Assistant via the Starling and HomeKit Controller.

**Comment by Monacoslo** [comment]
So what is now the solution, how to get Access Tooken now?

**Comment by iMicknl** [comment]
> Before I got into Home Assistant, I had gotten the Starling Home Hub and I was able to get the Protects into Home Assistant via the Starling and HomeKit Controller.

Depending on what they use, we could change this integration as well. Unfortunately, the most user-friendly way of generating a token does not seem supported anymore.

**Currently my time is very limited, thus I did not have time to look into this issue. Contributions to this integration are always welcome. Otherwise, we would just need to wait.** If you are able to generate a token in a different way, you can pass this in the integration as well.

**Comment by Monacoslo** [comment]
But are there any other options how to get token?

**Comment by iMicknl** [comment]
@Monacoslo @MaJonker let's move the asks / discussion to the Home Assistant Community Forum: https://community.home-assistant.io/t/support-for-nest-protect/310084/last. 

Unfortunately, Google changed their authentication mechanism, and this was a risk we knew from the beginning. The aim of this integration was to offer the most user-friendly way of authenticating to Nest server and including Nest Protect in our Home Assistant installation.

Let's discuss actual workarounds and investigating of the issue here.

**Comment by hoorna** [comment]
@bmdadu, I installed [Homebridge Docker](https://github.com/oznu/docker-homebridge) on a Synology NAS which is working for sure. Then I installed the plugin Homebridge Nest without a problem. In the plugin I have to choose a connection methode to Google. From the documentation I understand I have to use the "Google Account - Cookies" method as "Google Account - Refresh Token" is not working anymore. So far so good. But then?

With the Google Account - Cookies you have stayed logged in at Google. Logging out means that the connection will be broken and that you have to make a new connection again. What does that mean practically? Will Homebridge (in Docker) as well as the plugin stayed logged in in Google?

**Comment by bmdadu** [comment]
Yes, IT Will. As long as you dont logout.

**Comment by pwwukpw** [comment]
Getting same error following Google account password change. The append does not work .. any  ideas ? 

![Screenshot_20221110_193633_Samsung Internet](https://user-images.githubusercontent.com/88422402/201190909-57d13754-2735-455b-8a42-363499e2f5c3.jpg)

**Comment by Lazy8onez** [comment]
Did anyone find a solution for the authorization issue?
I am getting the "Access Blocked" error.
Appending "&ack_oob_shutdown=2022-10-03" does not help for me.

**Comment by JunctionArsonist** [comment]
> Did anyone find a solution for the authorization issue? I am getting the "Access Blocked" error. Appending "&ack_oob_shutdown=2022-10-03" does not help for me.

still broken...

**Comment by iMicknl** [comment]
See https://community.home-assistant.io/t/support-for-nest-protect/310084/169?u=imick.

**Comment by janusn** [comment]
@iMicknl Could you tell me which file to be modified in order to use the issue_token and the cookies retrieved from the method in the link?

Is it the configuration.yaml in the HomeAssistant configuration directory? Should it be written as follow?

```
nest_protect:
  googleAuth:
    issueToken: <issueToken here>
    cookies: <cookies here>
```

**Comment by IceBotYT** [comment]
> @iMicknl Could you tell me which file to be modified in order to use the issue_token and the cookies retrieved from the method in the link?
> 
> 
> 
> Is it the configuration.yaml in the HomeAssistant configuration directory? Should it be written as follow?
> 
> 
> 
> ```
> 
> nest_protect:
> 
>   googleAuth:
> 
>     issueToken: "<issueToken here>"
> 
>     cookies: "<cookies here>"
> 
> ```
> 
> 

I used the UI config flow and put it in there instead of configuration.yaml

**Comment by janusn** [comment]
@IceBotYT

Thanks for the quick reply. Could you enlighten me on how to convert issueToke and cookies into Access Token?


<img width="504" alt="Screen Shot 2022-11-25 at 3 40 46 pm" src="https://user-images.githubusercontent.com/21253783/204018583-b6609519-c4df-4374-9612-e811e5ae8ffe.png">

**Comment by IceBotYT** [comment]
You have the wrong repo added.

1. Go to the HACS tab.
2. Remove the old Nest Protect repo.
3. Press the ellipsis (...) menu in the top right.
4. Press custom repositories.
5. Paste in `https://github.com/nicoinch/ha-nest-protect` into the `Repository` field.
6. Set the `Category` field to `Integration`.
7. Click Add.
8. You will see a new Nest Protect integration in HACS. Download it.
9. Once downloaded, restart HA.
10. Start the config flow by clicking this My button: [![Open your Home Assistant instance and start setting up a new integration.](https://my.home-assistant.io/badges/config_flow_start.svg)](https://my.home-assistant.io/redirect/config_flow_start/?domain=nest_protect)
11. Paste in the issue token first, then the cookies.
12. You're done!

**Comment by janusn** [comment]
@IceBotYT 

Thanks a ton! It works now!

**Comment by MaJonker** [comment]
Dude you are a lifesaver - it finally works....! That costs me a few years of my life but THANKS for the clear explanation!

**Comment by danieldevos90** [comment]
Is there anyone who has the same problem as I have? I fill in the issue token and the cookies but I got this error message!

`Traceback (most recent call last):
  File "/config/custom_components/nest_protect/pynest/client.py", line 220, in authenticate
    self.nest_session = NestResponse(**nest_response)
TypeError: NestResponse.__init__() got an unexpected keyword argument 'error_description'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/nest_protect/config_flow.py", line 104, in async_step_account_link
    [issue_token, cookies] = await self.async_validate_input(user_input)
  File "/config/custom_components/nest_protect/config_flow.py", line 62, in async_validate_input
    await client.authenticate(
  File "/config/custom_components/nest_protect/pynest/client.py", line 227, in authenticate
    raise PynestException(
custom_components.nest_protect.pynest.exceptions.PynestException: 400 error while authenticating - {"error":"invalid_request","error_description":"missing user credentials","instance_id":"1dd43fa4-132b-486f-bcd2-001b6c9eeaa0"}. Please create an issue on GitHub.
`

**Comment by hammer-of-thor** [comment]
I used this a few hours ago and it worked  fine for me. Which guide did
use? To do the set-up?

**Comment by danieldevos90** [comment]
The flow of IceBotYT. And I did it in incognito mode.

**Comment by Erwin537** [comment]
> The flow of IceBotYT. And I did it in incognito mode.

For me incognito doesn't allow cookies so doesn't work. I used Edge instead.

**Comment by danieldevos90** [comment]
It is sooo weird, I got this issue as well on Brave and Safari..

**Comment by raym777** [comment]
> You have the wrong repo added.
> 
> 1. Go to the HACS tab.
> 2. Remove the old Nest Protect repo.
> 3. Press the ellipsis (...) menu in the top right.
> 4. Press custom repositories.
> 5. Paste in `https://github.com/nicoinch/ha-nest-protect` into the `Repository` field.
> 6. Set the `Category` field to `Integration`.
> 7. Click Add.
> 8. You will see a new Nest Protect integration in HACS. Download it.
> 9. Once downloaded, restart HA.
> 10. Start the config flow by clicking this My button: [![Open your Home Assistant instance and start setting up a new integration.](https://camo.githubusercontent.com/637fd24a458765d763a6dced4b312c503f54397bdd9b683584ef8054f305cd7f/68747470733a2f2f6d792e686f6d652d617373697374616e742e696f2f6261646765732f636f6e6669675f666c6f775f73746172742e737667)](https://my.home-assistant.io/redirect/config_flow_start/?domain=nest_protect)
> 11. Paste in the issue token first, then the cookies.
> 12. You're done!

Thank you very much for this. 

The only issue I'm noticing is that occupancy statuses of my smoke sensors don't appear to be updating. They have been stuck in "Detected" for 19 hours now (since my last HA restart). Anyone else?

**Comment by Lazy8onez** [comment]
I am probably missing something. Can someone please guide me on how to get the Issue_token and cookies? Thanks!

**Comment by JAudi23** [comment]
I followed these instructions using Chrome Incognito mode and it seems to be working for me:
https://github.com/nicoinch/ha-nest-protect

**Comment by Lazy8onez** [comment]
Perfect!
It worked for me on Edge.
Thanks!

**Comment by JamCrackersBV** [comment]
> 

Works for me as well

**Comment by sco0ts** [comment]
I now see "Please get your issue_token and cookies following the instructions in the integration README and paste them below." But I do not see instructions in the readme

**Comment by Lazy8onez** [comment]
The instructions are here:
https://github.com/nicoinch/ha-nest-protect

**Comment by SgtFoose** [comment]
> 5\. https://github.com/nicoinch/ha-nest-protect

This worked for me in Incognito mode! HINT: Use the popup "copy value", do not try to copy paste in notepad like I did. I tried this so many times, they all failed until I used copy value!

**Comment by nivek1612** [comment]
Worked for me a treat I now have Nest Protects fully integrated. The only minor wobble was as we have to separate homes segregated under one Gmail account the above process will import both homes' devices. Yes, first world problem. But good to be aware.

**Comment by chiversc** [comment]
nice job, works beautifully

**Comment by nivek1612** [comment]
Well I had to rebuild my HA long story, and now the token creation explained here no longer works for me 

https://github.com/nicoinch/ha-nest-protect

Get "Invalid authentication"

**Comment by acidoverflow** [comment]
Does not work for me. I get a Timeout when trying to add the integration

Logger: custom_components.nest_protect
Source: custom_components/nest_protect/pynest/client.py:168
Integration: Nest Protect ([documentation](https://github.com/imicknl/ha-nest-protect), [issues](https://github.com/imicknl/ha-nest-protect/issues))
First occurred: 09:08:05 (1 occurrences)
Last logged: 09:08:05

Traceback (most recent call last): File "/config/custom_components/nest_protect/config_flow.py", line 104, in async_step_account_link [issue_token, cookies] = await self.async_validate_input(user_input) File "/config/custom_components/nest_protect/config_flow.py", line 62, in async_validate_input await client.authenticate( File "/config/custom_components/nest_protect/pynest/client.py", line 168, in authenticate async with self.session.post( File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 1138, in __aenter__ self._resp = await self._coro File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 466, in _request with timer: File "/usr/local/lib/python3.10/site-packages/aiohttp/helpers.py", line 721, in __exit__ raise asyncio.TimeoutError from None asyncio.exceptions.TimeoutError

**Comment by croldan72** [comment]
I was getting this same error and attempted to follow the instructions to install the new version. While the new one is now working, it appears that I wasn't able to successfully delete the older version and now I also have 2 new ones as well. Can some one direct me to try to help clean up?

![image](https://user-images.githubusercontent.com/61128401/210249136-cb00030c-e985-4a1d-8a34-d9a93ff0d147.png)

**Comment by foothook** [comment]
Hi thanks all for the advice. It is not working for me, even if I tried to retrieve the cookie method as described.
I found the request-url and the cookie as described and saved them for using them for the HA login.

I also made sure to use the whole URL got from iframerpc 
But the configuration window refuses the url as access token, see screnshot:
![error_nest_protect_Screenshot 2023-01-08 193302](https://user-images.githubusercontent.com/122225362/211214934-be44e619-0f27-471d-b054-88ddf8e745f3.png)
I tried also to find errors in the url, like to fill the spaces in the url with %20 (....&scope=openid profile email https:...)

am I something missing - how do I provide my "issue_token" and "cookie" to the integration? can I put them manually in some configuration file?

please note: I have installed the integration by manually downloading \custom_components\nest_protect to my docker since the easy method by clicking the link does not work

**Comment by chinmayjade** [comment]
The cookie workaround method worked for me using Edge. Thanks folks!

**Comment by m4v3r1ckNl** [comment]
The cookie workaround method worked for me using Chrome incognito mode in macOS Mojave 10.14.6. Thanks to all dedicated en persistent developers/users of this great integration into Home Assistant. Very happy to have successfully adopted my 2 Nest Protect devices.

**Comment by mrlindstrom** [comment]
I was using nicoinch's fork with the new token method successfully. I added a new nest temperature sensor that wasn't showing up even after a reload/restart. I removed the integration and attempted to re-add it. It shows successful when I enter the token and cookie, but not devices or entities show up and the logs only show the below:

`023-01-16 21:09:40.895 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.nest_protect
2023-01-16 21:09:40.896 INFO (MainThread) [homeassistant.components.select] Setting up select.nest_protect
2023-01-16 21:09:40.896 INFO (MainThread) [homeassistant.components.switch] Setting up switch.nest_protect
2023-01-16 21:09:40.898 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up nest_protect platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 289, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/nest_protect/sensor.py", line 73, in async_setup_entry
    NestProtectSensor(device, description, data.areas, data.client)
  File "/config/custom_components/nest_protect/entity.py", line 118, in __init__
    super().__init__(bucket, description, areas, client)
  File "/config/custom_components/nest_protect/entity.py", line 33, in __init__
    self.area = areas[self.bucket.value["where_id"]]
KeyError: '00000000-0000-0000-0000-000100000008'`

**Comment by alexruffell** [comment]
I had the Nest Protects integrated for a long time using this integration then something happened and I just could not get it to work again... until I saw this comment which tipped me that I needed to install a package ahead of master:

https://github.com/iMicknl/ha-nest-protect/issues/105#issuecomment-1374909248

@foothook THANK YOU for pointing it out!

**Comment by MikeMannox** [comment]
I was just speculating, is this repo now dead as it seems we are being directed to use the [https://github.com/nicoinch/ha-nest-protect](https://github.com/nicoinch/ha-nest-protect) repo instead? 
I am reluctant to try using this "cookie method" as I feel this it is a workaround and is something too technical for some.

@iMicknl @nicoinch

**Comment by iMicknl** [comment]
@MikeMannox I wouldn't call it dead, I just have been very busy lately. There is still a wish to merge the new improvements and release a new version. I do own a Nest Protect (wired) myself as well, so I am still committed to have it working. (mine still works based on the previous auth method).

I fully agree that the cookie method is very technical and not user friendly at all, I have been looking in alternative ways to authenticate, but couldn't find any yet. The HomeBridge solution also moved back to the cookie method.

**Comment by MikeMannox** [comment]
@iMicknl Thanks for clarifying, and I would also like to thank you for your efforts with this task and project too, I know it can be difficult to maintain an OSS project.

I'll wait patiently until a valid solution can be found.

**Comment by B4dC4rd** [comment]
> I hope my edit helps prevent that others run into the same error as me
https://github.com/iMicknl/ha-nest-protect/issues/105#issuecomment-1374909248

Definitely did, cheers!

**Comment by mrlindstrom** [comment]
> I was using nicoinch's fork with the new token method successfully. I added a new nest temperature sensor that wasn't showing up even after a reload/restart. I removed the integration and attempted to re-add it. It shows successful when I enter the token and cookie, but no devices or entities show up and the logs only show the below:
> 
> `023-01-16 21:09:40.895 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.nest_protect 2023-01-16 21:09:40.896 INFO (MainThread) [homeassistant.components.select] Setting up select.nest_protect 2023-01-16 21:09:40.896 INFO (MainThread) [homeassistant.components.switch] Setting up switch.nest_protect 2023-01-16 21:09:40.898 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up nest_protect platform for sensor Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 289, in _async_setup_platform await asyncio.shield(task) File "/config/custom_components/nest_protect/sensor.py", line 73, in async_setup_entry NestProtectSensor(device, description, data.areas, data.client) File "/config/custom_components/nest_protect/entity.py", line 118, in __init__ super().__init__(bucket, description, areas, client) File "/config/custom_components/nest_protect/entity.py", line 33, in __init__ self.area = areas[self.bucket.value["where_id"]] KeyError: '00000000-0000-0000-0000-000100000008'`

I narrowed it down the the line referenced above:

`self.area = areas[self.bucket.value["where_id"]]`

if I edit that to:

`self.area = "Default"`

All my entities load without issue. any thoughts as to what it might be trying to set the area to that would make it completely error out? All of my entities are temperature sensors if that makes a difference.

**Comment by KurtaPajama** [comment]
The cookies method worked for me.

**Comment by alexruffell** [comment]
I used the method described in comment https://github.com/iMicknl/ha-nest-protect/issues/105#issuecomment-1374909248 to get my integration working again. Now I am getting an upgrade noticed stating I have 3.10 and that 3.11 is available. Since I had to download the custom component to fix my system I am wondering whether updating to 3.11 will break that.

**Comment by acidoverflow** [comment]
> 

Hi,

I tried it again today and it seems to be nearly successful but I still get an error even though the authentication returns valid:

`Logger: custom_components.nest_protect
Source: custom_components/nest_protect/pynest/client.py:161
Integration: nest_protect ([documentation](https://github.com/imicknl/ha-nest-protect), [issues](https://github.com/imicknl/ha-nest-protect/issues))
First occurred: 10:05:20 (2 occurrences)
Last logged: 10:20:58

GoogleAuthResponseForCookies.__init__() got an unexpected keyword argument 'valid'
Traceback (most recent call last):
  File "/config/custom_components/nest_protect/config_flow.py", line 104, in async_step_account_link
    [issue_token, cookies] = await self.async_validate_input(user_input)
  File "/config/custom_components/nest_protect/config_flow.py", line 54, in async_validate_input
    auth = await client.get_access_token_from_cookies(issue_token, cookies)
  File "/config/custom_components/nest_protect/pynest/client.py", line 161, in get_access_token_from_cookies
    self.auth = GoogleAuthResponseForCookies(**result)
TypeError: GoogleAuthResponseForCookies.__init__() got an unexpected keyword argument 'valid'`

Any idea on this? As I understand it GoogleAuthResponse returns "valid" but it's not what is expected to move on with the authentication process.

**Comment by Sany-IO** [comment]
You gave the Wrong extension!
There is a problem with HACS, i installed the other add-on with cookie login, but my hacs says there is a update available, when you click on update, HACS downloads THIS plugin, and not the plugin from the other author!!

**Comment by Swiftnesses** [comment]
I also updated and it broke my previous install (the patched version provided by @nicoinch.

Can we expect a fix here?

**Comment by iMicknl** [comment]
Not on very very short term since I am very busy. You can move back to the patched version of @nicoinch for now.

**Comment by alexruffell** [comment]
@iMicknl I am guessing your latest update fixes the 2023.3 issue (see error below), right? I had your integration installed and then manually copied over the custom_components from @nicoinch 's repository which makes it work but apparently only for a few more days (I am on a beta of 2023.3 and the integration still appears to work). I did not see any recent updates on his repo and there appears to be no way to submit issues so I am leaving it here just in case.

```
Logger: homeassistant.helpers.frame
Source: helpers/frame.py:77
First occurred: 08:51:22 (1 occurrences)
Last logged: 08:51:22

Detected integration that called async_setup_platforms instead of awaiting async_forward_entry_setups; this will fail in version 2023.3. Please report issue to the custom integration author for nest_protect using this method at custom_components/nest_protect/__init__.py, line 126: hass.config_entries.async_setup_platforms(entry, PLATFORMS)
```

or maybe it is actually not working already...
```
2023-02-27 10:01:47.607 ERROR (MainThread) [custom_components.nest_protect] Unknown exception. Please create an issue on GitHub with your logfile. Updates paused for 5 minutes.
Traceback (most recent call last):
File "/config/custom_components/nest_protect/__init__.py", line 162, in _async_subscribe_for_data
entry_data.client.nest_session = await entry_data.client.authenticate(auth)
File "/config/custom_components/nest_protect/pynest/client.py", line 191, in authenticate
"cookie": "G_ENABLED_IDPS=google; eu_cookie_accepted=1; viewer-volume=0.5; cztoken="
TypeError: can only concatenate str (not "NoneType") to str
```

**Comment by GaetanLabouche** [comment]
Hi,
I'm new to Home Assistant and therefore also to Github. I'm trying to integrate my Nest Protect and I'm having a problem authorizing my account with an error Error 400: invalid_request
Do you have a solution to help me knowing that I understand absolutely nothing in development
Thanks in advance and thanks for spending time developing the integration for dummies like me.

**Comment by tggman** [comment]
It's a little tricky but totally doable. Just  follow the instructions and use this forked  integration:

Nest Protect Integration Forked to use the Token/Cookies Method of Authentication:
https://github.com/nicoinch/ha-nest-protect

Instructions - Retrieving issue_token and cookies:
https://github.com/nicoinch/ha-nest-protect#retrieving-issue_token-and-cookies

Notes:
Last I checked, the integration needed a minor update but its still working for me. 

I did not use HACS but instead used the manual method of simply copying the appropriate folder from this repository into the 'custom_components' folder (create this folder if it doesn't exist) under the config folder of your Home Assistant install.

**Comment by tit1** [comment]
Thanks @tggman i used the other repo and worked perfectly.

**Comment by GaetanLabouche** [comment]
Hi,
Sorry to be so bad but I don't understand which part I have to change in the manual installation to be able to put the information to recover by having opened the browser in developer mode. I did find both pieces of information but I did not understand or integrate them.
Thank you for your patience and sorry I did not have time to try before

**Comment by JaneX8** [comment]
Unfortunately I can't get this intergration to work too.

**Comment by SMac3265** [comment]
I'm having a bit of the same problem as the two posts before mine. I've done the manual install of the files from the repository to my custom_components folder. Then I do the integration part, get asked to configure, use my "Google Account" ... and get the same error everyone else is having, where Nest is blocked. If the cookies & oauth method are supposed to solve that, I'm not really sure where they're supposed to be entered.

**Comment by tggman** [comment]
This integration use a different authentication method and needs to be removed and replaced with this integration:
https://github.com/nicoinch/ha-nest-protect

... which uses the issue_token and cookies method for authentication.

**Comment by SMac3265** [comment]
> This integration use a different authentication method and needs to be removed and replaced with this integration: https://github.com/nicoinch/ha-nest-protect
> 
> ... which uses the issue_token and cookies method for authentication.

OK...  I removed anything related to Nest Protect out of my HA. It no longer exists in HACS or in HA integrations itself. I also deleted everything I could from the Custom_Components subfolder (wouldn't let me remove pynest, though).

From there, I clicked on the link above - so to make sure I understand, that's a fork from the original, correct?

Once in there, I start the automatic HACS integration. Which when completed, wants me to authenticate with Nest/Google, and then... same issue as before. Need a token that I cannot get, unless it's buried in the long oauth2 string that I picked up from following a different suggestion in here.

Nothing was added to the custom_components folder when doing the installation. Does that mean I need to manually add the files again?

I'm not purposely trying to be difficult or dense, and I'm not stupid with technology... but so far, this whole process is maddening. 

Sean

**Comment by eliotharper** [comment]
@SMac3265 you need to use the forked repo provided by @tggman — follow these steps carefully, pasting the `issue_token` and `cookies` values in the two provided input fields when you install this forked version (the fields don't have labels). [I can help](https://calendly.com/eliotharper/30min) if you get stuck.

**Comment by MikeMannox** [comment]
@iMicknl 
Is anyone able to give an update on how close this project is to resolving the issue? (I am not referring to hacking tokens & what-not)
I wonder would ChatGPT 4.0 or GitHub co-pilot be able to help? (Serious question)

2 months have passed since I posted my last comment, I saw one update on HACS which seemed to be unrelated to this issue.

cc. @iMicknl

**Comment by iMicknl** [comment]
@MikeMannox no, unfortunately not. Google is not open and they don't provide a public API for the Nest Protect. The earlier method has been blocked by recent app changes and thus at the moment the only possibility is to use the cookie method.

I don't have much time at the moment to work on my opensource projects, however there have been some great contributions on this repository to enable the cookie method as a work around. Happy to merge them in when the current issues / bugs are fixed :).

**Comment by quincarter** [comment]
> OK... I removed anything related to Nest Protect out of my HA. It no longer exists in HACS or in HA integrations itself. I also deleted everything I could from the Custom_Components subfolder (wouldn't let me remove pynest, though).

I was able to get this working utilizing this fork just now. If you are techy at all (which if you're using HA and you ended up here, then you probably are), it isn't too hard to setup. I had to add this forked repo in my HACS custom repositories, then i was able to add it through HACS without having to manually add the files. 

Can anyone verify how long the session/cookie/token lasts? Will i have to perform another incognito login like once a month or does this forked integration take care of this for me? Just wondering how much upkeep i will have to do.

**Comment by iMicknl** [comment]
Please update to latest pre-release (beta) via HACS or GitHub releases and try again, if you cannot use the old authentication method.

**Comment by alexruffell** [comment]
@iMicknl - are you saying your version now works with the cookie method? I am using the nicoinch version. Can I now click "update" and overwrite it with yours?

**Comment by iMicknl** [comment]
Yes. Make sure to update to the pre-release version (0.4.0b1).

**Comment by alexruffell** [comment]
@iMicknl While it appears to be working, I am getting authentication errors in my logs reported here:

https://github.com/iMicknl/ha-nest-protect/issues/221

**Comment by hesspoint** [comment]
Any rough ETA when the stable release with cookie authentication will come. Just asking to gauge if it makes sense to go for the beta or wait for the stable release.
My track record with beta versions (not this repo but in general) in HA has not been great.

**Comment by iMicknl** [comment]
@hesspoint not soon, if you cannot use the other method, best is to switch to the prerelease.

**Comment by prashmohan** [comment]
In the latest pre-release it appears that the URL is not being generated. I get the error:
Translation Error: The intl string context variable "url" was not provided to the string "To link your Google account, [authorize your account](https://<myhostname>/config/integrations/dashboard).

After authorization, copy-paste the provided Auth Token code below."

**Comment by ejpenney** [comment]
I'm also getting the url generation error. Can't authenticate today.

**Comment by raym777** [comment]
I've been using the "retrieve issue token and cookies" method for a while and has always worked but after the latest 2023.06.x release of HA, I immediately started to see "Failed to setup" for the Nest Protect integration. I've tried re-installing the component and re-entering my token and cookies several times and it just will no longer authenticate for me...

**Comment by iMicknl** [comment]
@raym777 you are probably using a fork. Have a look at your custom repositories settings in HACS, remove the other repo and install the latest beta from here.

**Comment by raym777** [comment]
> @raym777 you are probably using a fork. Have a look at your custom repositories settings in HACS, remove the other repo and install the latest beta from here.

@iMicknl Thank you so much! That was it! Appreciate the help.

**Comment by chiversc** [comment]
I had the same problem, but installing beta version solved it. Thanks for the great work @iMicknl

**Comment by bitbyter9** [comment]
Ok, so I am in the process of installing the beta, I have gone through the token request process (I have the issue_token and the cookie) but don't know how to proceed after this point. When I go to integrations in HA and try to add Nest Protect, I select Google Account and a window opens asking me to input...something? I've tried my Gmail account address, the token and the cookie. Nothing seems to work. Any assistance as to what I'm missing would be appreciated.

**Comment by Buster14** [comment]
Hi, I tried both v0.4.0b2 and v0.4.0b1 but when clicking "authorize account" it sends me to HA integration page.
The latest stable version gave me error 400, what am i missing?

**Comment by ahmadnassri** [comment]
@Buster14 did you try to restart HA after install?

**Comment by Buster14** [comment]
Hi, i forgot how but i managed to get it working, but i decided not to use it since its not accurate enough to pickup smoke from my neighbours burning trash

**Comment by lantranbot** [comment]
Working for me as well with the beta version.  Thanks a bunch.

**Comment by stefanuytterhoeven** [comment]
Stil the 'blocked' error. I see here a few solutions, but they don't work for me

**Comment by iMicknl** [comment]
@stefanuytterhoeven did you use the latest beta version? Where you add your cookies manually.

**Comment by iainp999** [comment]
Hey there, I'm unable to make any progress on this at all. There seems to be a lot of information in the history of this issue - I've tried following the different approaches but still get the "Access blocked" error.

Is there a comprehensive set of current instructions that can get this going?

**Comment by Trembler34** [comment]
What am I missing here? The beta doesnt work either, the URL of the "Authorize Your Account" just redirects back to my HA Homepage.. How do we get this working with newest beta release?

**Comment by iMicknl** [comment]
@Trembler34 in the beta you should use cookies, not "Authorize your account". How did you install the beta?

**Comment by Trembler34** [comment]
> @Trembler34 in the beta you should use cookies, not "Authorize your account". How did you install the beta?

What do you mean by Use Cookies? Only thing I see is to use Authorize Your Account then a section I am assuming to paste an authorization code? I installed it through HACS..

**Comment by Trembler34** [comment]
here is what i see right now, latest beta release. 
![Screenshot 2023-12-14 114117](https://github.com/iMicknl/ha-nest-protect/assets/37017861/b30c2c1d-1e67-4702-950e-a9ef0d691c48)

**Comment by iMicknl** [comment]
@Trembler34 could you try the latest version? https://github.com/iMicknl/ha-nest-protect/releases/tag/v0.4.0b3. Please double check if you are not using a fork.

**Comment by Trembler34** [comment]
Did you just publich that new version? It wasnt there when I installed it a bit ago prior to this post. Installing it now.

**Comment by Trembler34** [comment]
Here is the one Im using, just installed B3 and it still looks the same. Re-downloaded,restarted HA. 
![Screenshot 2023-12-14 114117](https://github.com/iMicknl/ha-nest-protect/assets/37017861/fe1d814d-663f-4c5e-88a9-65f7dd0fc86b)
![Screenshot 2023-12-14 122244](https://github.com/iMicknl/ha-nest-protect/assets/37017861/fb5b4902-7fd9-4a7f-b766-b2a26e4cc5a5)
![Screenshot 2023-12-14 122320](https://github.com/iMicknl/ha-nest-protect/assets/37017861/cafea6e0-a239-4494-9dcb-bf68c8266696)

**Comment by iMicknl** [comment]
Can you download it manually? Remove `nest_protect` from your custom_components folder and download the zip from the link above. Then place `nest_protect` in your custom_components folder. 

The text on your error is not part of the code anymore, so I don't know where it comes from tbh..

**Comment by Trembler34** [comment]
ok, let me give it a try..

**Comment by Trembler34** [comment]
Thats better, let me go through the README and see how it works. It seems the one downloading from HACS is pulling an old file or something. 
![Screenshot 2023-12-14 114117](https://github.com/iMicknl/ha-nest-protect/assets/37017861/7acab04c-d6a4-43ab-a7fe-03165cde1cb6)

**Comment by Trembler34** [comment]
sweet, it worked! Finally I have all my puck temps throughout the house.. One question, what is the downside to this? It seems it may be a bit sesntive and tied to the computer that orginally logged into the browser with? What are some of the gotchas? Appreciate the quick help with this..

**Comment by Trembler34** [comment]
Looks like it already stopped working. Is this kind of the expectation with this kind of setup using cookies on a browser and relying on that account staying signed into the browser? or is there going to be some way to make sure this stays more consistently connected?  
![Screenshot 2023-12-15 094741](https://github.com/iMicknl/ha-nest-protect/assets/37017861/27a77d6b-9809-4494-b587-03e98ceb2f8b)

**Comment by iMicknl** [comment]
@Trembler34 let's discuss this in a new issue if you still face issues, and keep this issue open for mor general discussion. Best is to use Incognite mode and the instructions from https://github.com/iMicknl/ha-nest-protect/issues/272.

**Comment by Trembler34** [comment]
I set it up per the instructions. Worked fine for 24hrs then died. I’m of the understanding that this is just kind of how it is with this current method. I decided to set it up again from a chrome browser that runs on a server that never gets touched and so far it’s been more solid. At this point I am good. Will keep an eye out for changes in the future. Thanks for the follow up.

**Comment by Technically-Possible** [comment]
im struggling to get past this part I can't choose a different version to download I don't know what to do from here any advice 
![image](https://github.com/iMicknl/ha-nest-protect/assets/48217923/384a9549-e9b6-4154-9662-1fb850b5f2dc)

**Comment by iMicknl** [comment]
@Technically-Possible you need the beta version. Select "use pre-release" in HACS, or download beta version from GitHub.

**Comment by Technically-Possible** [comment]
im not able to find it in HACS I was trying to install it though the integration tab in home assistant. how can I add it to HACS as its not currently there

**Comment by Technically-Possible** [comment]
![image](https://github.com/iMicknl/ha-nest-protect/assets/48217923/69df5351-65c7-4ed3-9e2d-fbe2d648f479)

**Comment by Technically-Possible** [comment]
my cameras are accessible under https://home.google.com will this process of getting the cookies still work in this case

**Comment by Technically-Possible** [comment]
![image](https://github.com/iMicknl/ha-nest-protect/assets/48217923/e176a472-37f7-401b-ab38-3c73ce178280)
i assume this is because the cams are only able to be seen on https://home.google.com/ not home.nest

**Comment by iMicknl** [comment]
@Technically-Possible this integration is for the Nest Protect smoke alarm. We don’t support the cameras.

**Comment by techfixpros** [comment]
Please add to the instructions to enable third party cookies in incognito mode. Just spent an hour trying to figure out why none of the calls were showing up in the pane.

**Comment by tonyrsutton** [comment]
So, will this ever be fixed or not?

Thanks.

**Comment by hesspoint** [comment]
Broken again after latest update 2024.2.3.
Failed Setup will retry.

Tried to add entry and entered my credentials as per the instructions and it says already setup.

**Comment by techfixpros** [comment]
> Please add to the instructions to enable third party cookies in incognito mode. Just spent an hour trying to figure out why none of the calls were showing up in the pane.

NOTE: The current beta is working properly for over 24 hours now. I would assume this issue can be closed as soon as the beta is merged to main.

**Comment by tonyrsutton** [comment]
Good news. Any ideas when that will happen?

Thanks for the update.

**Comment by hesspoint** [comment]
I just checked have the latest beta v0.4.0b4. Even re-downloaded it.

Still broken

**Comment by iMicknl** [comment]
@hesspoint please create a new issue and define what 'still broken' means. If you face the same problem as described in this issue, you definitely aren't running the beta version.

**Comment by hesspoint** [comment]
> @hesspoint please create a new issue and define what 'still broken' means. If you face the same problem as described in this issue, you definitely aren't running the beta version.

Well yes and no. I am running the beta for quite a while not and could connect using the issue token and cookie. That was working well for quite some time. With the latest update of HA 2024.2.3 it suddenly stopped working and says Failed Setup will retry. Also restart won't fix it.

I tried to click "add entry" and I grabbed and pasted the issue token and cookie again (as per instructions). Then I got the error message that an entry already exists and that's it, Still does not work

**Comment by nota3k** [comment]
Seeing the same issue as @hesspoint . Updated to latest beta, was on b1 before but it broke with some HA update.

**Comment by iMicknl** [comment]
Please create a new issue with more details and your log.

**Comment by B-V-** [comment]
Setting this integration for the first time. 
My finding is that there isn't any ' issueToken' found when logged in home.nest.com.
Is Google using a different token config?
Will start a new issue for this with all the details.

![Google_Nest_Protect_Issue_No_issueToken](https://github.com/iMicknl/ha-nest-protect/assets/6977812/04ccd267-e0ab-43d2-8eb2-dfc1d9bdba68)

**Comment by adambullmer** [comment]
Seems as of 2023-01-31 the workaround posted above doesn't work anymore and this app probably needs to migrate its auth instructions: https://developers.google.com/identity/protocols/oauth2/resources/oob-migration

**Comment by iMicknl** [comment]
@adambullmer that is discussed extensively in this thread already. You need to use the beta for a new auth method (which is not great, but the only one working).

**Comment by adambullmer** [comment]
oh! Missed the year timestamp on the comments changing and didn't see this had been commented post shutoff 🤦‍♂️. I appreciate the clarification. I think the switch up from the OB acknowledge workaround and to the beta version (using a new auth flow) got lost in github's comment rollup, and I had thought the beta was just for appending the query param automatically. I'll give the beta a whirl

**Comment by aafonstein** [comment]
Hi, trying to setup this intergration, but I cannot get to the page for authentication, it redirects me back to HA,

When I try to add it, in the screen it gives me the error 
Translation Error: The intl string context variable "url" was not provided to the string

is there something i'm missing here?

**Comment by hamzah** [comment]
@iMicknl It might be worth letting people know to use the beta somewhere in the README to avoid any duplicate issues in the meantime.

**Comment by bharat** [comment]
Hey team. I'm getting a localization error when I use 4.0b5. I'm new to HomeAssistant - I only installed it to get this particular plugin (thanks, Mick!).

HA 2024.6.3
HA Frontend: 20240610.1
HACS: 1.34.0

![image](https://github.com/iMicknl/ha-nest-protect/assets/54570/b90c786c-169f-496e-b065-962376b5bc57)

I followed the instructions from the README to get the issueToken and cookies and was able to proceed.

**Comment by iMicknl** [comment]
@bharat how do you install the integration? Please first remove your nest_protect folder from custom_components. Afterwards add the latest version (4.0.b5) in the custom_components folder.
If you still face issues, please create a new issue.

**Comment by bharat** [comment]
Thanks for the fast response. I removed the integration via the web UI, restarted HA, reinstall the the latest `main` integration, then immediately switched to `4.0.b5`, and restarted HA again. This time around it worked... not sure what I did last time, but I was learning HA as I was going so I might have skipped a step without realizing it. FWIW, I'm running inside a Docker container and other than checking what's in the `custom_components` directory I didn't touch anything in there so somehow I got into that state via the UI, but unfortunately I can't reproduce it at this point.

Separately, I did get hung up a bit on logging into Nest in my Chrome incognito window without enabling 3rd party cookies. It took me a moment to figure that out. Would you like me to submit a PR to add that step to the documentation?

**Comment by iMicknl** [comment]
@bharat how did you install the integration at first? This error is caused by older files that are still present on your system. So it seems that with an update, only new files have been added, existing files haven't been deleted.

Regarding contributions, there are already a few open PRs with doc improvements. If you can have a look at them and see if anything is missing, a PR is always appreciated! Please make sure to use the beta branch and target beta.

**Comment by bharat** [comment]
Unfortunately, this was my first go-round with HomeAssist, HACS, and ha-nest-protect so I don't have a clear recollection of all the steps that I took to get it working since there were a lot of false starts. If I can reproduce it, I'll let you know!

**Comment by kiler129** [comment]
FYI: at present the Nest login is completely broken so (re)authorizing isn't possible. It goes into a login loop on Google's side with `Error: Nest scope is missing` being logged in dev console. They completely broke their own OAuth2 flow... maybe finally they're experimenting with moving away from Protect being the only lost child in Nest.com ecosystem?

https://www.googlenestcommunity.com/t5/Nest-Thermostats/web-oauth-login-infinite-loop/m-p/626500

**Comment by cgoudie** [comment]
> Hey team. I'm getting a localization error when I use 4.0b5. I'm new to HomeAssistant - I only installed it to get this particular plugin (thanks, Mick!).
> 
> HA 2024.6.3 HA Frontend: 20240610.1 HACS: 1.34.0
> 
> ![image](https://private-user-images.githubusercontent.com/54570/340123184-b90c786c-169f-496e-b065-962376b5bc57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1MDMyNDAsIm5iZiI6MTcxOTUwMjk0MCwicGF0aCI6Ii81NDU3MC8zNDAxMjMxODQtYjkwYzc4NmMtMTY5Zi00OTZlLWIwNjUtOTYyMzc2YjViYzU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI3VDE1NDIyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRkYmEyYjg3YzA2ZmZiODIxNjU0MWMyZWQ2YzFhMDJjNGU5ODA0MzFiMTkwMTFjZTc1Yjg1MjBkM2ZmNmY4ZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Kkk7JgRP5ASrShmyD9DdX4GdUDqOQuwMdRe4HTuP5pA)
> 
> I followed the instructions from the README to get the issueToken and cookies and was able to proceed.

Today, this is the only way I was able to make it work, install the beta, read the readme from the beta branch ( https://github.com/iMicknl/ha-nest-protect/blob/beta/README.md ) and then copy the two things needed into the form. It then connected and worked.

**Comment by Naldjer** [comment]
Tried the method from the readme, but i dont know in which configuration form i have to paste it. Anyone could point me out please?

**Comment by rollac01** [comment]
I don't suppose there has been any fix for this issue? I have just downloaded this integration through HACS for the first time and at this stage I am unable to generate an access token. Thanks.

**Comment by iMicknl** [comment]
@Naldjer, @rollac01 you have to download the beta version. The current stable version doesn't work for new accounts.

**Comment by Naldjer** [comment]
@iMicknl  many thanks! Got it working now! :)

**Comment by Trembler34** [comment]
I am finally getting around to checking why all my stuff got disconnected. I noticed in my intergrations it says user logged out. I went through the readme again but can't seem to fine the IssueToken anymore. I'm sure I missed something or some process has changed its been awhile since I set this up.. I figured the readme would be updated.. What is the new process to get this working?

**Comment by simonotter** [comment]
Could anyone tell me how to install the beta version in HACS to use the working authentication method, please?

**Comment by dfensom** [comment]
> Could anyone tell me how to install the beta version in HACS to use the working authentication method, please?
[
](https://community.home-assistant.io/t/integrating-nest-protect-smoke-alarms/603253)

**Comment by dfensom** [comment]
https://community.home-assistant.io/t/integrating-nest-protect-smoke-alarms/603253

**Comment by CaptainLexington** [comment]
I'm not an expert in auth methods, but I think they have changed their API. Trying this just now on Opera, I had nothing in the network tab filtered with `issueToken`. But filtering just for `issue`, I get `https://nestauthproxyservice-pa.googleapis.com/v1/issue_jwt`, which is a POST instead of a GET, and the auth token is in the request payload rather than the URI. Obviously, just copying this URL into the first box does not work. `iframe` request with the cookie still seems to work, although it doesn't seem to have the `rpc` substring anymore.

**Comment by TheShai1** [comment]
Couple things; 
1. Was told to download the latest Beta Release. None of the downloadable versions in HACS say Beta. The most recent "Pre-Release" version v0.4.0b7 is the version I had installed and did nothing. I tried rolling back to the previous version and got the same result. 
2. In HA there is no network tab in Developer Tools; therefore there is no "Preserve Log" to check and no "Filter" box to enter "issue token". 
3. One network call (beginning with iframerpc) does not appear in the Dev Tools window, so we cant click on it.
4.  No headers tab under General so there is no issue token. 
Can't complete any of this integration instructions except opening home.nest.com

**Comment by TheShai1** [comment]
I still keep getting the same error from Google Nest. But now that I know you mean the developer tools in chrome I'll have to try that. 
I can report in a new issue. I went to this one as the other one was closed. 
I understand still having bugs before releasing official version plus Google can be a PITA but I do appreciate the effort as I've wanted to incorporate my nest protects in.

**Comment by TheShai1** [comment]
@iMicknl Thank you!!!! I just tried it and it worked!!!! It doesn't show the actual devices with entities underneath but it does show all the entities. 
Thank you for the quick response and your efforts.

**Comment by iMicknl** [comment]
@TheShai1 feel free to comment / do a pull request on what can be improved in our documentation.

**Comment by iMicknl** [comment]
Personally I am still on the old authentication method... I understood that the tokens should actually live more than a few weeks usually. Did you start the session in private mode?

**Comment by smholsen** [comment]
> Personally I am still on the old authentication method... I understood that the tokens should actually live more than a few weeks usually. Did you start the session in private mode?

I've actually tried both incognito and without. Not sure if the cause is that the token is "forgotten" in HA on restarts, or something else. 
I've thought about creating an automation in HA that checks the status of the integration, and performs the token retrieval and re-setup of the integration when it becomes unavailable. But I don't really want to store my pw in HA secret, and also I guess it would be tricky with 2fa. Perhaps it is possible to create some sort of app-token or something I could use 🤔 
Would be curious to hear if you have other ideas that might be better for this?

**Comment by iMicknl** [comment]
Home Assistant doesn't forget the token, a re-login means that the session is not valid anymore. And automating this process will be hard I am afraid, the previous authentication method was great, but got killed by Google... Still hoping that someday someone will figure out / reverse engineer their current login mechanism.

**Comment by RobinTTY** [comment]
Thanks for all your work with this integration. Would just recommend to add a note to the README that the beta version is required for it to work and maybe reference this issue.
